[Build]: MacOS Build fails in CI

### Operating System

MacOS in our CI

### Compiler

Xcode 16.1 / 15.4

### ChakraCore Version

main

### Compiler output

```shell
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/functional:521:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/__functional/bind.h:20:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/tuple:1455:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/exception:82:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/__exception/exception_ptr.h:16:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/__memory/construct_at.h:23:
/Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/new:285:35: error: redefinition of 'operator delete'
  285 | inline _LIBCPP_HIDE_FROM_ABI void operator delete(void*, void*) _NOEXCEPT {}
      |                                   ^
/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Common/Memory/Allocator.h:466:1: note: previous definition is here
  466 | operator delete(
      | ^
2 errors generated.
[193/540] Building CXX object lib/Parser/CMakeFiles/Chakra.Parser.dir/CharSet.cpp.o
FAILED: lib/Parser/CMakeFiles/Chakra.Parser.dir/CharSet.cpp.o 
/usr/bin/clang++ -DBIT64=1 -DCHAKRA_STATIC_LIBRARY=1 -DCLANG_HAS_DISABLE_TAIL_CALLS=1 -DDISABLE_JIT=1 -DFEATURE_PAL -DHAS_ICU -DHAS_REAL_ICU=1 -DINTL_ICU=1 -DNO_PAL_MINMAX -DPAL_STDCPP_COMPAT -DPLATFORM_UNIX -DPLATFORM_UNIX=1 -DSTACK_ALIGN=16 -DTARGET_64 -DUNICODE -D_ARM64_ -D_ARM64_=1 -D_CHAKRACOREBUILD -D_M_ARM32_OR_ARM64 -D_M_ARM64 -D_SAFECRT_USE_CPP_OVERLOADS=1 -D__STDC_WANT_LIB_EXT1__=1 -D__aarch64__ -D__arm64__=1 -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/. -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Common -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Common/PlaceHolder -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/pal -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/pal/inc -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/pal/inc/rt -I/opt/homebrew/opt/icu4c/include -I/opt/homebrew/opt/icu4c/include/../i18n -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../JITIDL -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Common -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Backend -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime -I/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime/ByteCode -isystem /usr/local/include -mmacosx-version-min=10.9 -std=gnu++11         -Wno-ignored-attributes        -Wno-parentheses-equality        -Wno-reorder        -Wno-microsoft        -Wno-unused-value        -Wno-int-to-void-pointer-cast        -Wno-invalid-offsetof        -Wno-undefined-inline        -Wno-inconsistent-missing-override        -Wno-c++14-extensions        -Wno-macro-redefined        -Wno-pragmas        -Wno-invalid-token-paste        -Wno-format        -Wno-invalid-noreturn        -Wno-null-arithmetic        -Wno-tautological-constant-out-of-range-compare        -Wno-tautological-undefined-compare        -Wno-address-of-temporary        -Wno-null-conversion        -Wno-return-type        -Wno-switch        -Wno-int-to-pointer-cast        -Wno-tautological-constant-compare        -Wno-enum-compare-switch        -Wno-unknown-warning-option         -Werror         -Wno-error                  -fno-omit-frame-pointer         -fdelayed-template-parsing -O3 -DNDEBUG -std=gnu++11 -arch arm64 -isysroot /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk -mmacosx-version-min=14.6   -fdiagnostics-color=always -arch arm64 -mmacosx-version-min=10.9 -fasm-blocks -fms-extensions -fwrapv -fno-rtti -fPIC -MD -MT lib/Parser/CMakeFiles/Chakra.Parser.dir/CharSet.cpp.o -MF lib/Parser/CMakeFiles/Chakra.Parser.dir/CharSet.cpp.o.d -o lib/Parser/CMakeFiles/Chakra.Parser.dir/CharSet.cpp.o -c /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/CharSet.cpp
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/CharSet.cpp:5:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/ParserPch.h:22:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime/Runtime.h:334:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime/PlatformAgnostic/ChakraPlatform.h:7:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime/PlatformAgnostic/UnicodeText.h:8:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime/PlatformAgnostic/ChakraICU.h:38:
In file included from /opt/homebrew/opt/icu4c/include/unicode/ucol.h:1524:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/functional:521:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/__functional/bind.h:20:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/tuple:1455:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/exception:82:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/__exception/exception_ptr.h:16:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/__memory/construct_at.h:23:
/Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/new:279:66: error: redefinition of 'operator new'
  279 | _LIBCPP_NODISCARD_AFTER_CXX17 inline _LIBCPP_HIDE_FROM_ABI void* operator new(std::size_t, void* __p) _NOEXCEPT {
      |                                                                  ^
/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Common/Memory/Allocator.h:457:1: note: previous definition is here
  457 | operator new(
      | ^
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/CharSet.cpp:5:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/ParserPch.h:22:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime/Runtime.h:334:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime/PlatformAgnostic/ChakraPlatform.h:7:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime/PlatformAgnostic/UnicodeText.h:8:
In file included from /private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Parser/../Runtime/PlatformAgnostic/ChakraICU.h:38:
In file included from /opt/homebrew/opt/icu4c/include/unicode/ucol.h:1524:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/functional:521:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/__functional/bind.h:20:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/tuple:1455:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/exception:82:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/__exception/exception_ptr.h:16:
In file included from /Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/__memory/construct_at.h:23:
/Applications/Xcode_16.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX15.1.sdk/usr/include/c++/v1/new:285:35: error: redefinition of 'operator delete'
  285 | inline _LIBCPP_HIDE_FROM_ABI void operator delete(void*, void*) _NOEXCEPT {}
      |                                   ^
/private/var/folders/xf/_tm0f94d66n8kr12tqwrylrr0000gn/T/cirrus-ci-build/lib/Common/Memory/Allocator.h:466:1: note: previous definition is here
  466 | operator delete(
      | ^
2 errors generated.
ninja: build stopped: subcommand failed.
```


### Additional Context

https://github.com/chakra-core/ChakraCore/runs/34919774427

Azure: [Online Logs](https://dev.azure.com/ChakraCoreEngine/ChakraCore/_build/results?buildId=850&view=logs&j=75d718b2-3687-5f91-e9bc-b9a9c7cb5b82&t=5b7e928e-e355-5a1f-8010-2562fdbbf977) [Raw Download](https://github.com/user-attachments/files/18293174/azure-build-8edf0ea9-cbea-40ed-97a1-84f5086324ec.txt)

11a685147600551b2361f04daec57dec2566a2ec
4398657e948c0deb1887b9af54ca7aab37349d20

@ppenzin

--- a/lib/Common/Memory/Allocator.h
+++ b/lib/Common/Memory/Allocator.h
@@ -1,5 +1,6 @@
 //-------------------------------------------------------------------------------------------------------
 // Copyright (C) Microsoft. All rights reserved.
+// Copyright (c) ChakraCore Project Contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
 //-------------------------------------------------------------------------------------------------------
 #pragma once
@@ -448,6 +449,8 @@ void AssertValue(void * mem, T value, uint byteCount)
 #define NO_EXPORT(x) x
 #endif
 
+#if defined(_MSC_VER) && !defined(__clang__)
+
 // For the debugger extension, we don't need the placement news
 #ifndef __PLACEMENT_NEW_INLINE
 #define __PLACEMENT_NEW_INLINE
@@ -470,6 +473,14 @@ void * previousAllocation               // Previously allocated memory
 {
 
 }
+
+#endif
+
+#else
+
+// Use std inline placement new instead of custom
+// See PR #7009
+#include <new>
 
 #endif
 

